src/main/js/wiki/components/BranchDropdown.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import injectSheet from "react-jss";
+import classNames from "classnames";
+import { orderBranches } from "../../repository/components/BranchOverview";
+
+const styles = {
+  dropdown: {
+    "background-color": "white",
+    border: "#777777",
+    color: "#777777",
+    padding: "6px 12px",
+    fontSize: "16px",
+    marginLeft: "auto"
+  }
+};
+
+export type Branch = {
+  name: string;
+  link: string;
+};
+
+type Props = {
+  branches: Branch[];
+  currentBranch: string;
+  classes: any;
+};
+
+export const buildBranchUrl = (oldBranchName: string, newBranchName: string) => {
+  const url = window.location.href;
+  return url.replace(oldBranchName, newBranchName);
+};
+
+class BranchDropdown extends React.Component<Props> {
+  constructor(props) {
+    super(props);
+    console.log(props);
+    this.state = {
+      repository: props.repository,
```

Comment:
I think `repository` and `branches` should not be part of the state since they are not managed by the component.

Reply:
That's right, the state is managed in the 'page' container. At least for the 'currentBranch' however I set the state also in this component (when switching branches). My problem here is that we are using class components and I can't use hooks like 'useState' in them.
Since I need to use the state variables, what would you suggest to solve this problem?

Reply:
replaces state with props